diff --git plugins/resources/unixfilesystem/libunixfilesystem.cpp plugins/resources/unixfilesystem/libunixfilesystem.cpp
index 190fabe07..59132c0fb 100644
--- plugins/resources/unixfilesystem/libunixfilesystem.cpp
+++ plugins/resources/unixfilesystem/libunixfilesystem.cpp
@@ -468,7 +468,7 @@ void warn_if_deprecated_context_string_set(
             __FUNCTION__);
     }
 
-    const std::vector<const std::string> deprecated_keys{HIGH_WATER_MARK, REQUIRED_FREE_INODES_FOR_CREATE};
+    const std::vector<std::string> deprecated_keys{HIGH_WATER_MARK, REQUIRED_FREE_INODES_FOR_CREATE};
 
     for (const auto& k : deprecated_keys) {
         std::string holder{};
