diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7235017..6a313b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(IRODS_PACKAGE_REVISION "1")
 
 include(RequireOutOfSourceBuild)
 
-set(CMAKE_CXX_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang++)
+# set(CMAKE_CXX_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang++)
 set(CMAKE_CXX_STANDARD 17)
 # export-dynamic so stacktrace entries from executables have function names
 set(CMAKE_EXE_LINKER_FLAGS_INIT "-rdynamic -Wl,--export-dynamic -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
@@ -17,6 +17,10 @@ project(icommands
   VERSION "${IRODS_VERSION}"
   LANGUAGES CXX)
 
+include(GNUInstallDirs)
+include_directories(AFTER ${CMAKE_INSTALL_PREFIX}/include)
+include_directories(${IRODS_EXTERNALS_FULLPATH_JSON}/include/nlohmann)
+
 set(CMAKE_SKIP_BUILD_RPATH OFF)
 set(CMAKE_SKIP_INSTALL_RPATH OFF)
 set(CMAKE_SKIP_RPATH OFF)
@@ -24,8 +28,6 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF)
 set(CMAKE_BUILD_RPATH_USE_ORIGIN ON)
 
-include(UseLibCXX)
-
 set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
 
 include(${IRODS_TARGETS_PATH})
@@ -462,3 +464,5 @@ if (NOT CPACK_GENERATOR)
 endif()
 
 include(CPack)
+
+install(TARGETS ${IRODS_CLIENT_ICOMMANDS_EXECUTABLES} DESTINATION ${CMAKE_INSTALL_BINDIR})
