diff --git a/lib/core/include/irods/irods_load_plugin.hpp b/lib/core/include/irods/irods_load_plugin.hpp
index 1e6bed3cf..f8ce24a70 100644
--- a/lib/core/include/irods/irods_load_plugin.hpp
+++ b/lib/core/include/irods/irods_load_plugin.hpp
@@ -165,6 +165,15 @@ namespace irods
             msg += "]";
             return ERROR( PLUGIN_ERROR_MISSING_SHARED_OBJECT, msg );
         }
+	
+        // Defensively remove NUL from the constructed so_name
+        // string. We have observed a NUL appear between the directory
+        // and file name components of the so_name path when
+        // distributed as a Conda package.
+        //
+        // When c_str() is called on the offending string the result
+        // is truncated and dlopen() on the truncated path fails.
+        so_name.erase(remove(so_name.begin(), so_name.end(), '\0'), so_name.end());
 
         // =-=-=-=-=-=-=-
         // try to open the shared object
