diff --git a/plugins/api/src/data_object_finalize.cpp b/plugins/api/src/data_object_finalize.cpp
index 59d99e2d7..e8263a86f 100644
--- a/plugins/api/src/data_object_finalize.cpp
+++ b/plugins/api/src/data_object_finalize.cpp
@@ -106,7 +106,7 @@ namespace
 
     auto column_is_mutable(const std::string_view _c)
     {
-        using immutable_columns_list_t = std::vector<const std::string_view>;
+        using immutable_columns_list_t = std::vector<std::string_view>;
         static const immutable_columns_list_t immutable_columns =
         {
             "data_id",
