diff --git a/S3/ACL.py b/S3/ACL.py
index 175725a8..7f4d2455 100644
--- a/S3/ACL.py
+++ b/S3/ACL.py
@@ -186,9 +186,9 @@ def revoke(self, name, permission):
         permission = permission.upper()
 
         if "ALL" == permission:
-            self.grantees = [g for g in self.grantees if not (g.name.lower() == name or g.display_name.lower() == name)]
+            self.grantees = [g for g in self.grantees if not (g.name.lower() == name or (g.display_name is not None and g.display_name.lower() == name))]
         else:
-            self.grantees = [g for g in self.grantees if not ((g.display_name.lower() == name or g.name.lower() == name)
+            self.grantees = [g for g in self.grantees if not (((g.display_name is not None and g.display_name.lower() == name) or g.name.lower() == name)
                 and g.permission.upper() == permission)]
 
     def get_printable_tree(self):
