diff --git a/iRODS/config/platform.mk.template b/iRODS/config/platform.mk.template
index fa9fa20..a72c224 100644
--- a/iRODS/config/platform.mk.template
+++ b/iRODS/config/platform.mk.template
@@ -196,17 +196,17 @@ endif
 
 LDADD += -lcurl
 
-MY_CFLAG+= -I/usr/include/openssl
+# MY_CFLAG+=-I/usr/include/openssl
 ifeq ($(OS_platform), linux_platform)
 LDADD+=-L/usr/lib/x86_64-linux-gnu/
 CL_LDADD+=-L/usr/lib/x86_64-linux-gnu/
 endif
 
 # added by Ilari Korhonen and Niclas Jareborg (for OSX 10.11+)
-CCFLAGS+=-I/usr/local/opt/openssl/include
-MY_CFLAG+=-I/usr/local/opt/openssl/include
-LDADD+=-L/usr/local/opt/openssl/lib
-CL_LDADD+=-L/usr/local/opt/openssl/lib
+CCFLAGS+=-I$(PREFIX)/include
+#MY_CFLAG+=-I/usr/local/opt/openssl/include
+LDADD+=-L$(PREFIX)/lib
+#CL_LDADD+=-L/usr/local/opt/openssl/lib
 
 LDADD+=-lssl -lcrypto
 CL_LDADD+=-lssl -lcrypto
diff --git a/iRODS/lib/core/src/irods_server_properties.cpp b/iRODS/lib/core/src/irods_server_properties.cpp
index e279440..9235e90 100644
--- a/iRODS/lib/core/src/irods_server_properties.cpp
+++ b/iRODS/lib/core/src/irods_server_properties.cpp
@@ -138,7 +138,7 @@ namespace irods {
 
         buf[BUF_LEN - 1] = '\0';
         fchar = fgets( buf, BUF_LEN - 1, fptr );
-        for ( ; fchar != '\0'; ) {
+        for ( ; fchar != NULL; ) {
             if ( buf[0] == '#' || buf[0] == '/' ) {
                 buf[0] = '\0'; /* Comment line, ignore */
             }
diff --git a/packaging/build.sh b/packaging/build.sh
index af90e27..0250e79 100755
--- a/packaging/build.sh
+++ b/packaging/build.sh
@@ -335,7 +335,7 @@ confirm_preflight_prerequisites() {
             echo "      :: NOT A DETECTED OPERATING SYSTEM" 1>&2
         fi
         echo "#######################################################${text_reset}" 1>&2
-        exit 1
+        # exit 1
     fi
 
     if [ "$PREFLIGHTDOWNLOAD" != "" ] ; then
@@ -343,7 +343,7 @@ confirm_preflight_prerequisites() {
         echo "ERROR :: $SCRIPTNAME requires some software to be installed" 1>&2
         echo "$PREFLIGHTDOWNLOAD" 1>&2
         echo "#######################################################${text_reset}" 1>&2
-        exit 1
+        # exit 1
     fi
 
     # print out python prerequisites error
